examples/low-level/nft.js

```diff
@@ -1,51 +1,18 @@
+import * as near from '../../src/api'
+
 const MAX_SUPPLY = 10
 const TOTAL_SUPPLY = 'c'
 
 export function mint_to() {
-    let tokenId
-    let hasRead = env.jsvm_storage_read(TOTAL_SUPPLY, 0)
-    if (hasRead != 0) {
-        tokenId = Number(env.read_register(0))+1
-    } else {
-        tokenId = 1
-    }
+    let tokenId = near.jsvmStorageRead(TOTAL_SUPPLY)
+    tokenId = tokenId ? Number(tokenId) + 1 : 1
     if (tokenId > MAX_SUPPLY) {
         throw new Error('Maximum token limit reached')
     }
-    env.jsvm_args(0)
-    let input = env.read_register(0)
-    let parsedInput = JSON.parse(input)
-    let owner_id = parsedInput['owner']
-    env.jsvm_storage_write(TOTAL_SUPPLY, tokenId.toString(), 0)
-    env.jsvm_storage_write('token_to_owner_' + tokenId, owner_id, 0)
-    env.log(`Minted NFT ${tokenId} to ${owner_id}`)
-    env.jsvm_value_return(tokenId.toString())
-}
-
-
-// What we want, a user friendly high-level API like this one:
-// * class based
-// * arguments and return are auto converted from env.input() and env.value_return
-// * class properties are auto load and save to blockchain state
-/*
-@NearBindgen
-class MyNFT {
-    constructor(maxSupply) {
-        this.maxSupply = maxSupply
-        this.totalSupply = 0
-        this.tokenOwner = new LookupMap('token_to_owner_')
-    }
-
-    mintTo(owner) {
-        if (this.totalSupply < this.maxSupply) {
-            this.totalSupply++
-        } else {
-            throw new Error('Maximum token limit reached')
-        }
-        let tokenId = this.totalSupply;
-        this.tokenOwner.insert(owner, tokenId);
-        env.log(`Minted NFT ${tokenId} to ${ownerId}`)
-        return tokenId
-    }
+    let input = near.jsvmArgs();
+    let ownerId = input['owner']
+    near.jsvmStorageWrite(TOTAL_SUPPLY, tokenId.toString())
+    near.jsvmStorageWrite('token_to_owner_' + tokenId, ownerId)
+    near.log(`Minted NFT ${tokenId} to ${ownerId}`)
+    // env.jsvm_value_return(tokenId.toString()) //TODO: Not implemented. Should we add it?
```

Comment:
> @ailisp I'm not sure what should be done with this function call. The documentation says that it is used together with env.jsvm_call(...args), but we do not have it here.

My bad on describing the behavior. jsvm_value_return does: 1) store the return in a C static variable, so if this were called by another contract, that contract can retrieve this info. 2) plus the normal host function value_return.

This contract doesn't call other contract so there's no jsvm_call, but this contract could be called by another contract. Therefore, for any JS contract, jsvm_value_return should always be used instead of value_return. 


> In this function we are using env.read_register(...args), not env.jsvm_value_return().

`env.jsvm_call` saves the the return (if there were jsvm_value_return in the call) in both a C static variable and the given register:

```
// this is the env.jsvm_call
static JSValue near_jsvm_call(JSContext *ctx, JSValueConst this_val, int argc, JSValueConst *argv) {
  // ...
  jsvm_call(contract_name_len, (char *)contract_name_ptr, method_name_len, (char *)method_name_ptr, arguments_len, (char *)arguments_ptr, true); // inside jsvm_call, if there's a jsvm_value_return, it saves call return to a static variable and also call host's value_return
  jsvm_call_returns(&call_ret, &ret_len, GET); // get the return from that static variable
  write_register(register_id, ret_len, (uint64_t)call_ret); // save to given register
  // ...
}
```

If we just want to return the value to runtime, `jsvm_value_return(value)` would do that. If we want to further process this value, we need to read it from register (e.g. in `jsvmCall`)